tsx/README.md

```diff
@@ -0,0 +1,44 @@
+# `@11ty/eleventy-plugin-tsx`
+
+Adds support for TSX template files in Eleventy v3.0 and newer.
+
+- 11ty Docs https://www.11ty.dev/docs/languages/typescript/
+- `tsx` package: https://tsx.is
+- `jsx-async-runtime` package: https://github.com/jeasx/jsx-async-runtime
+
+## Installation
+
+```sh
+npm install @11ty/eleventy-plugin-tsx
+```
+
+Add to your configuration file (ESM version shown):
+
+```js
+import tsxPlugin from "@11ty/eleventy-plugin-tsx";
+
+export default function (eleventyConfig) {
+	eleventyConfig.addPlugin(tsxPlugin);
+}
+```
+
+## How it works
+
+Eleventy template files ending in `.11ty.tsx` will be processed by this plugin, by means of the [`tsx` project](https://tsx.is). `tsx` turns the template into JSX nodes using the [`esbuild`](https://esbuild.github.io) package.
```

Comment:
Why do they have to end in `.11ty.tsx`? Seems like an odd constraint...

Reply:
The `.11ty` part is required by 11ty. The `.tsx` is connected to `tsconfig.json` and the compiler. While it might be possible to configure the toolchain to accept another file extension, it would be very atypical. 

Either way, it's the choice made by this plugin. Kind of matches other template language plugins, which provide default mappings.

Reply:
Why is it required? I've never used anything but plain `.tsx` extensions before, so it seems like Eleventy is the only blocker here?

Reply:
Right. Look in the "File Extension" column in the [current docs](https://www.11ty.dev/docs/languages/jsx/). This is related to their derivation from [JavaScript Templates](https://www.11ty.dev/docs/languages/javascript/).

11ty thinks these JSX/TSX templates are actually just JavaScript Templates with a different `compile` property.

Reply:
Ahh gotcha. And there isn't a way to get Eleventy to think like that *and* remove the `11ty.` part?

Reply:
It's a derivation of JavaScript Templates and it follows what Zach has for existing JSX/TSX documentation. 

If you'd like, I could tinker with changing the line `key: "11ty.js"` in `tsxConfig.js`. Perhaps it isn't important for people to think of these as JavaScript Template types.